### ISSUE TITLE
Support Event logs for no asset definition tokens

### ISSUE THREAD

**Issue by oa-s** [AlphaWallet/alpha-wallet-ios#5698]
Can we make fetching events for tokens that has no asset definition, applying Event filter based on token type, and its abi?
**Reason**:  Some of tokens has no `asset definition`, e.g most of erc1155 tokens,  and we are not able to see its events on activity tab, my suggestion is to use token abi, for retrieving event definition and fetch such event logs with `getEventLogs` fetcher.

**Comment by oa-s** [comment]
@hboon  any ideas according to it?

**Comment by hboon** [comment]
So asset definition here refers to TokenScript, right?

> Can we make fetching events for tokens that has no asset definition, applying Event filter based on token type, and its abi?

But we already do this right?

The 2 classes with this header:

> ///Fetching ERC1155 tokens in 2 steps:
> ///
> ///A. Fetch known contracts and tokenIds owned (now or previously) for each, writing them to JSON. tokenIds are never removed (so we can easily discover their balance is 0 in the next step)
> ///B. Fetch balance for each tokenId owned (now or previously. For the latter value would be 0)
> ///
> ///This class performs 

> Reason: Some of tokens has no asset definition, e.g most of erc1155 tokens, and we are not able to see its events on activity tab, my suggestion is to use token abi, for retrieving event definition and fetch such event logs with getEventLogs fetcher.

Or do you mean that we pull the events to discover ERC1155 contracts and token IDs but not for activity?

**Comment by oa-s** [comment]
> But we already do this right?
yes, but we fetch event logs for tokens that already have token script files, but not for those who don't have, so we could update our code base with assumption to fetch event logs based on erc type for each token (that don't have token script definitions).

**Comment by oa-s** [comment]
if you check activity tab for erc1155 token, you will see that there is no events.

**Comment by hboon** [comment]
Hmm.. Yeah I guess we need a parallel to `EventSourceForActivities` or update it to support ERC1155 OR maybe better to save the events in `Erc1155TokenIdsFetcher`.

Let's hold this for a bit unless it's really easy (updating `Erc1155TokenIdsFetcher` is probably easy?). Need to think a bit about the wallet API and see if it's getting anywhere to avoid doubling the work.

Do you think updating `Erc1155TokenIdsFetcher` will work?

**Comment by oa-s** [comment]
Do you think updating Erc1155TokenIdsFetcher will work?

> yes, could work

**Comment by hboon** [comment]
@oa-s ok, feel free to try your hand at updating `Erc1155TokenIdsFetcher` then
